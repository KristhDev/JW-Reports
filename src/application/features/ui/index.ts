export { default as uiReducer, UI_INITIAL_STATE, setIsDataExporting, setIsKeyboardVisible, setOldDatetimePicker } from './slice';
export * from './types';