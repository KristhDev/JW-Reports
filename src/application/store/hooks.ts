import { useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
export const useAppSelector = useSelector.withTypes<RootState>();