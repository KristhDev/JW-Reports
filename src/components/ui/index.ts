export { Button } from './Button';
export { EyeBtn } from './EyeBtn';
export { FormField } from './FormField';
export { Title } from './Title';