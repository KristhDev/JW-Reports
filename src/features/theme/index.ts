export { default as themeReducer, setTheme } from './slice';