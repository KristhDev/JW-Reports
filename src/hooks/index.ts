export { default as useAuth } from './useAuth';
export { default as useTheme } from './useTheme';