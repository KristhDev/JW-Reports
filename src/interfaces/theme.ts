export type Theme = 'dark' | 'light' | 'default';

/* Defining the interface for the ThemeState. */
export interface ThemeState {
    colors: Colors;
    deviceTheme: Theme;
    isLoadedTheme: boolean;
    selectedTheme: Theme,
    theme: Theme,
}

/* Defining the interface for the Colors object. */
export interface Colors {
    background: string;
    bottom: string;
    button: string;
    buttonDark: string;
    card: string;
    contentHeader: string;
    focus: string;
    header: string;
    headerText: string;
    icon: string;
    inputText: string;
    linkText: string;
    modal: string;
    modalText: string;
    navbar: string;
    text: string;
    titleSecondary: string;
    titleText: string;
}

/**
 * SetThemePayload is an object with a property called theme that is of type Theme
 * @property {Theme} theme - The theme to set.
 */
export type SetThemePayload = {
    theme: Theme
}