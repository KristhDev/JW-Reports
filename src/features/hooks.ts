import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch: () => AppDispatch = useDispatch;
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;