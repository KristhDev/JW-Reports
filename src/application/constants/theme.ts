import { ItemOption } from '../../infrasturcture/interfaces';

/**
 * The theme options to select
 */
export const THEME_OPTIONS: ItemOption[] = [
    { label: 'Modo predeterminado', value: 'default' },
    { label: 'Modo claro', value: 'light' },
    { label: 'Modo oscuro', value: 'dark' }
];