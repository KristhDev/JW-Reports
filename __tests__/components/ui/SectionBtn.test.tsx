import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react-native';

/* Components */
import { SectionBtn } from '../../../src/components/ui';

/* Hooks */
import { useTheme } from '../../../src/hooks';

/* Theme */
import { darkColors } from '../../../src/theme';

const text = 'Section text test';
const subText = 'Section sub text test';
const onPressMock = jest.fn();

/* Mock hooks */
jest.mock('../../../src/hooks/useTheme.ts');

describe('Test in <SectionBtn /> component', () => {
    (useTheme as jest.Mock).mockReturnValue({
        state: { colors: darkColors },
        BUTTON_TRANSPARENT_COLOR: 'rgba(255, 255, 255, 0.15)'
    });

    beforeEach(() => {
        render(
            <SectionBtn
                onPress={ onPressMock }
                subText={ subText }
                text={ text }
            />
        );

        jest.clearAllMocks();
    });

    it('should to match the snapshot', () => {
        expect(screen.toJSON()).toMatchSnapshot();
    });

    it('should render respective props', () => {

        /* Get elements with props of component */
        const textElement = screen.getByTestId('section-btn-text');
        const subTextElement = screen.getByTestId('section-btn-sub-text');

        /* Check if elemets exists and containt props */
        expect(textElement).toBeTruthy();
        expect(textElement.props.children).toBe(text);
        expect(subTextElement).toBeTruthy();
        expect(subTextElement.props.children).toBe(subText);
    });

    it('should call onPress when pressed', () => {

        /* Get touchable */
        const touchable = screen.getByTestId('section-btn-touchable');
        fireEvent.press(touchable);

        /* Check if onPress is called one time */
        expect(onPressMock).toHaveBeenCalledTimes(1);
    });
});