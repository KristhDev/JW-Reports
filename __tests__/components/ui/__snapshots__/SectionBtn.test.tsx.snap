// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <SectionBtn /> component should to match the snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  centered={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "overflow": "hidden",
      },
      {
        "padding": 16,
      },
    ]
  }
  testID="section-btn-touchable"
>
  <View>
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontSize": 16,
        }
      }
      testID="section-btn-text"
    >
      Section text test
    </Text>
    <Text
      style={
        {
          "color": "#BFBFBF",
          "fontSize": 14,
        }
      }
      testID="section-btn-sub-text"
    >
      Section sub text test
    </Text>
  </View>
</View>
`;
